README.md

```diff
@@ -5,16 +5,75 @@ This is the main repository related to the Meson project.
+These are the basics for joining the Meson mixnet as a provider, authority or mix. For more in-depth documentation on the Meson project, visit out [project site](https://hashcloak.com/Meson).
 
-### How to run a provider
-TODO
+## Usage
 
-### How to run a mix
-TODO
+#### How to Run a Provider or Mix Node
 
-### How to join the Authority PKI
-TODO
+
+All of our infrastructure uses docker setups. You will first need to generate a provider config and its PKI keys. The easiest way to do that is by using our [genconfig](https://github.com/hashcloak/genconfig/#genconfig) script:
+
+```bash
+go get github.com/hashcloak/genconfig
+genconfig \
+  -a 138.197.57.19 \ # Current ip address of authority
+  -authID RJWGWCjof2GLLhekd6KsvN+LvHq9sxgcpra/J59/X8A= \ # Current public key of authority
+  -name provider-name \ # Your provider name
+  -ipv4 1.1.1.1 \ # Your public ipv4 address
+  -provider \ # Flag to indicate you only want a provider config
+```
+
+This will make a directory called `output/provider-name` with a file called `identity.public.pem`. Send us your public key to our email [info@hashcloak.com](info@hashcloak.com). We will then help you to get added to the mixnet as a provider. Once you give is your public key you can get your node running with:
+
+```bash
+docker service create \
+  --name meson -d \
+  -p 30001:30001 \ # Mixnet port
+  -p 40001:40001 \ # User registration port
+  --mount type=bind,source=`pwd`/output/provider-name,destination=/conf \
+  hashcloak/meson:master
+```
+
+__Note__ You will have to wait for about 10 minutes before your node is being used in the mixnet. It has to wait for a [new epoch](https://hashcloak.com/Meson/docs/#waiting-for-epoch).
+
+To run a mix node please take a look at the [docs](https://hashcloak.com/Meson/docs/#running-meson).
+
+#### How to send transactions
+
+__⚠️ WARNING ⚠️__: The mixnet is not ready for strong anonymity since it is still being worked on. The privacy features are not ready for production use. There is currently is support for both `Goerli` and `Rinkeby` testnets but most Ethereum based chain can be [supported](https://hashcloak.com/Meson/docs/#other-blockchains) by Meson.
+
+```bash
+# Clone the github repo with the demo wallet
+git clone https://github.com/hashcloak/Meson-wallet-demo
+cd Meson-wallet-demo
+
+RAW_TXN=0xf8640284540be40083030d409400b1c66f34d680cb8bf82c64dcc1f39be5d6e77501802ca0c434f4d4b894b7cce2d880c250f7a67e4ef64cf0a921e3e4859219dff7b086fda0375a6195e221be77afda1d7c9e7d91bf39845065e9c56f7b5154e077a1ef8a77
+go run ./cmd/wallet/main.go \
+  -t gor \ # gor is the ethereum chain ticker for the goerli testnet
+  -s gor \ # Meson service name
+  -rt $RAW_TXN \ # Signed raw transaction blob
+  -c client.toml \ # Config file
+  -chain 5 \ # Chain id for goerli
+```
+
+The contents of `client.toml` are:
```

Comment:
Explain what the client.toml file is in more details. Add comments explaining the parameters, etc. Maybe not in the readme.